apm-agent-plugins/apm-quartz-job-plugin/src/test/java/co/elastic/apm/agent/quartz/job/JobTransactionNameInstrumentationTest.java

```diff
@@ -0,0 +1,222 @@
+ * You may obtain a copy of the License at
+ * 
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ * #L%
+ */
+package co.elastic.apm.agent.quartz.job;
+
+import java.util.Collections;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import org.junit.BeforeClass;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Disabled;
+import org.junit.jupiter.api.Test;
+import org.quartz.Job;
+import org.quartz.JobBuilder;
+import org.quartz.JobDetail;
+import org.quartz.JobExecutionContext;
+import org.quartz.JobExecutionException;
+import org.quartz.Scheduler;
+import org.quartz.SchedulerException;
+import org.quartz.SchedulerFactory;
+import org.quartz.SimpleScheduleBuilder;
+import org.quartz.Trigger;
+import org.quartz.TriggerBuilder;
+import org.quartz.impl.JobDetailImpl;
+import org.quartz.impl.JobExecutionContextImpl;
+import org.quartz.impl.StdSchedulerFactory;
+
+import co.elastic.apm.agent.MockReporter;
+import co.elastic.apm.agent.bci.ElasticApmAgent;
+import co.elastic.apm.agent.configuration.SpyConfiguration;
+import co.elastic.apm.agent.impl.ElasticApmTracer;
+import co.elastic.apm.agent.impl.ElasticApmTracerBuilder;
+import co.elastic.apm.agent.quartz.job.JobTransactionNameInstrumentation;
+import co.elastic.apm.agent.report.ApmServerReporter;
+import net.bytebuddy.agent.ByteBuddyAgent;
+import org.springframework.scheduling.quartz.QuartzJobBean;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+
+class JobTransactionNameInstrumentationTest {
+
+    private static MockReporter reporter;
+    private static ElasticApmTracer tracer;
+
+    @BeforeClass
+    @BeforeAll
+    static void setUpAll() {
+        reporter = new MockReporter();
+        tracer = new ElasticApmTracerBuilder()
+                .configurationRegistry(SpyConfiguration.createSpyConfig())
+                .reporter(reporter)
+                .build();
+        ElasticApmAgent.initInstrumentation(tracer, ByteBuddyAgent.install(),
+                Collections.singletonList(new JobTransactionNameInstrumentation(tracer)));
+    }
+
+    @Test
+    void testJobWithGroup() throws SchedulerException, InterruptedException {
+        reporter.reset();
+        resetCounter();
+        Scheduler scheduler=new StdSchedulerFactory().getScheduler();
+        scheduler.clear();
+        JobDetail job = JobBuilder.newJob(TestJob.class)
+    			.withIdentity("dummyJobName", "group1").build();
+        Trigger trigger = TriggerBuilder
+        		.newTrigger()
+        		.withIdentity("myTrigger")
+        		.withSchedule(
+        				SimpleScheduleBuilder.repeatSecondlyForTotalCount(2, 1))
```

Comment:
Not sure if we need to test this, as it tests the framework usage and not the instrumentation.

If you think this is important to test, it should be enough to do it only in one dedicated test called `testRepeatedInvocations`. I assume the other things tested shouldn't matter with the repetition, right?

The problem is that this makes the unit tests much slower. As we accumulate tests, it is important for us to keep them as quick as possible. 

Unless there is a good reason, please remove the repetitions from all the tests and use only one without a counter, but with an explicit expected number of transactions.